### ISSUE TITLE
I'm not able to modify the doc object during the onBeforeCreate or onBeforeUpdate dynamic hooks

### ISSUE THREAD

**Issue by iamlogiq** [orientechnologies/orientdb#5147]
I wrote a javascript function that looks like that
```
print('data_onBeforeCreate doc (before) ---> ' + doc);
doc.field('created', '2014-02-07T00:02:00.000')
print('doc (after) ---> ' + doc);
```
I created the trigger on a class called data and when I create a new instance of data i see the console message telling me that the doc object has been modified with the new 'created' property. I even see the new 'created' field appear in the query response but when I look at the persisted instance (select from data where) I don't see the new 'created' property. 

I guess my question is...can you add new properties to the doc object before it is created?

**Comment by Ziink** [comment]
Add
```Javascript
return 'RECORD_CHANGED';
```
at the end of the function and see if that helps.

**Comment by iamlogiq** [comment]
@Ziink 
thank you..it worked but I'm having issues with i'm using an embedded list instead
```
doc.field('updated', [{'@type': 'd', '@class': 'access', 'id': '1234567', 'timeStamp': '2015-02-07T00:02:00.000'}])
return 'RECORD_CHANGED'
```
I get an error saying that I didn't provide an embedded list. How do I pass an embedded set or embedded list into the doc field? Better yet...do you know a link with examples of using dynamic hooks & functions where you actually have to modify the doc object. I was also having trouble calling an SQL function inside a javascript function. These topics are not well documented in the orientdb docs...is there better documentation out there?

**Comment by Ziink** [comment]
You might want to mark this as a documentation bug.

Try something like
```javascript
var OType = com.orientechnologies.orient.core.metadata.schema.OType;
doc.field('updated', listObj, [OType.EMBEDDEDLIST])
```
Note that the third argument is an array.

See [OType](http://orientdb.com/javadoc/latest/com/orientechnologies/orient/core/metadata/schema/OType.html) for other data types.

**Comment by iamlogiq** [comment]
@Ziink 
is the second argument (listObj) my actual data? I"m not familiar with java but did you mean something like this?
```
var OType = com.orientechnologies.orient.core.metadata.schema.OType;
var listObj = {'@type': 'd', '@class': 'accessStamp', 'timeStamp': '1967-02-07T00:02:00.000'};
doc.field('updated', listObj, [OType.EMBEDDEDLIST])
```

**Comment by Ziink** [comment]
Yes. That was Javascript code :) It works for me for EMBEDDEDSET. Did you try it?

**Comment by iamlogiq** [comment]
@Ziink 
It doesn't work for me... this is the code i'm running:
```
print('\\ndata_onBeforeCreate doc (before) ---> ' + doc);
var OType = com.orientechnologies.orient.core.metadata.schema.OType;
var listObj = {'@type': 'd', '@class': 'accessStamp', 'id': '1234567', 'timeStamp': '1967-02-07T00:02:00.000'};
doc.field('updated', listObj, [OType.EMBEDDEDLIST])
print('\\ndoc (after) ---> ' + doc);
return 'RECORD_CHANGED'
```
When I look at the console value for doc.updated (after) : 
```
updated:[4]
```
and I get this error: 
```
SEVER Internal server error:
com.orientechnologies.orient.core.exception.OValidationException: The field 'data.updated' has been declared as EMBEDDEDLIST but an incompatible type is used.
```
Shouldn't the console message for doc.updated look more like listObj instead of [4]?

**Comment by Ziink** [comment]
Try using an array instead, like 
```javascript
doc.field('updated', [listObj], [OType.EMBEDDEDLIST])
```

If that doesn't work, I have no clue how to make it work.

**Comment by iamlogiq** [comment]
Doesn't work but thank for the help!
